internal/controller/vrg_recipe.go

```
@@ -330,6 +338,89 @@ func recipeNamespacesValidate(recipeElements util.RecipeElements, vrg ramen.Volu
 	return nil
 }
 
+func recipeVMBackupValidate(ctx context.Context, reader client.Reader, recipeElements util.RecipeElements,
+	vrg ramen.VolumeReplicationGroup,
+	ramenConfig ramen.RamenConfig, log logr.Logger,
+) error {
+	vmProtection := isProtectingVirtualMachine(recipeElements, vrg, log)
+
+	if !vmProtection {
+		return nil
+	}
+
+	if ramenConfig.KubeObjectProtection.VeleroNamespaceName != "" ||
+		len(ramenConfig.KubeObjectProtection.VeleroNamespaceName) > 0 {
+		if isKubeVirtEnabled(ctx, reader, ramenConfig, log) {
+			return nil
+		}
+
+		return fmt.Errorf("kubevirt plugin is disabled, hence VM resources backup/restore might not succeed")
```

Need some clarification here: the error message indicates “backup/restore might not succeed”. Is failing the recipe validation the right response (i.e., refusing to proceed) or should this be logged as a warning allowing operation to continue? If the plugin absence means VM protection definitely will fail, failing early is correct. But if it’s only a “may fail” scenario, maybe a warning + skip path is better.